lib/Backend/BackwardPass.cpp

```diff
@@ -2107,6 +2107,16 @@ BackwardPass::DeadStoreTypeCheckBailOut(IR::Instr * instr)
         return;
     }
 
+    // If oldBailoutKind is equivTypeCheck and appropriate opnd doesn't have initial/final type
+    // then leave alone the bailout
+    if (oldBailOutKind == IR::BailOutFailedEquivalentTypeCheck || oldBailOutKind == IR::BailOutFailedEquivalentFixedFieldTypeCheck)
+    {
+        if (!(propertySymOpnd->HasInitialType() || propertySymOpnd->HasFinalType()))
```

Comment:
So I was mistakenly checking the `bailoutKind` that we have in forward pass instead of checking that for backward pass. With that, I don't need the check for initial/final type.